Bump @docusaurus/preset-classic from 2.1.0 to 2.2.0 (#19)
flipt-io/docs: authentication/using-tokens.mdx

---

> title: Using Client Tokens

Reviewer: what about just merging this page under the `Token` section in the `Method` page above?

Reviewer 2: My thought process here was that "client tokens" aren't strictly tied to a single method.
`token` is obviously the most likely method to leverage this, given it is just a static instantiation of a client token.

However, OIDC establishes a client token in the same way and the result can be used via both means too (`Authorization` or `Cookie` header). 

What do you think? This probably doesn't provide much practical value at this stage. We might add another method in the future which could leverage this.

Reviewer: yeah thats a good point, fine to keep as is. I'm just thinking that when I saw `Tokens` and `Client Tokens` they seemed like they should be under the same section. Maybe we need to think about how we differentiate the two more? 

Maybe changing the notion of `Token` (created via the API/bootstrap) to `Static Token` would make it more clear? (http://localhost:3000/authentication/methods#token).

Something we do think about/do in a separate PR tho as this one is g2g IMO!

Reviewer 2: 💯 I think you're right. The `token` method should be clearly titled `Static Tokens` to help differentiate how the token is acquired from using it on the wire.

> description: This document explains how handle client tokens via both HTTP and gRPC.

Reviewer: ```suggestion
description: This document explains how to handle client tokens via both HTTP and gRPC.
```

---

## HTTP

There are two different valid ways to present a client token via a HTTP requests.
The choice allows us to support two different types of workload.

### 1. `Authorization` Header

> For application to Flipt communication over HTTP, the `Authorization` header is most appropriate.

Reviewer: ```suggestion
For applications that communicate with Flipt over HTTP, the `Authorization` header is most appropriate.
```

    Name:  "flipt_client_token",
    Value: "gt6P_zIqTnCngfHDCpWb48ob5EBt3PqunUhpofNCNnc=",
  })

  resp, err := http.Do(req)
  // ...
}
```

```typescript client.ts
import fetch from 'node-fetch';

const headers = { 'Cookie': 'flipt_client_token=gt6P_zIqTnCngfHDCpWb48ob5EBt3PqunUhpofNCNnc=' }
const response = await fetch('https://flipt.your.instance/api/v1/flags', { headers: headers })
```

```python client.py
import requests

def doRequest():

  headers ={"Cookie": "flipt_client_token=gt6P_zIqTnCngfHDCpWb48ob5EBt3PqunUhpofNCNnc="}
  requests.get("https://flipt.your.instance/api/v1/flags", headers=headers)

  return
```
</CodeGroup>

This allows for stateful browser sessions to be established.

When using a "session compatible" authentication method (e.g. [OIDC](/authentication/methods#oidc)), Flipt will automatically established this cookie via a `Set-Cookie` response header during the authentication method exchange.

In a browser context this means subsequent API calls will be automatically authenticated given the API requests are invoked with credentials included (cookies are enabled). Flipt's own UI leverages this mechanism for it's login functionality.

## gRPC

> For gRPC we use the [Metadata](https://grpc.io/docs/what-is-grpc/core-concepts/#metadata) functionality similarly to HTTP Headers.

Reviewer: ```suggestion
For gRPC we use the [Metadata](https://grpc.io/docs/what-is-grpc/core-concepts/#metadata) functionality similar to HTTP Headers.
```
